enclave-runtime/src/rpc/author/mod.rs

```diff
@@ -119,15 +125,20 @@ where
 /// some unique operations via RPC and have them included in the pool.
 const TX_SOURCE: TrustedOperationSource = TrustedOperationSource::External;
 
-//impl<P, Client> AuthorApi<TxHash<P>, BlockHash<P>> for Author<P, Client>
-impl<'a, P> AuthorApi<TxHash<P>, BlockHash<P>> for Author<'a, P>
+enum TopSubmissionMode {
+	Submit,
+	SubmitWatch,
+}
+
+impl<'a, P> Author<'a, P>
 where
 	P: TrustedOperationPool + Sync + Send + 'static,
 {
-	fn submit_top(
+	fn process_top(
```

Comment:
extracted a common method to process tops (both `submit` and `submit_and_watch`), removes duplicate code. Uses an `enum` to distinguish between the modes, which is not the cleanest of designs, but it gets the job done. Tried it with function pointers first, but I got stuck there really quickly and couldn't find a simple solution.